### ISSUE TITLE
PHP-Insecure Random (PEN test vlunerability)

### ISSUE THREAD

**Issue by vjaykoogu** [ADOdb/ADOdb#1020]
### Description
We found vlunerability in PEN test results.

### Environment
- ADOdb version: 5.22.7
- Driver or Module: ADOdb
- PHP version: 8.1
- Platform: Linux

### Error found in below files and Method to Change.
![image](https://github.com/ADOdb/ADOdb/assets/87112/c94515e3-e951-406a-a7b2-a8510ea5faad)

- In above Screenshot we see filename and methos to change.

### Fix
Using Below solution:
To generate secure random numbers in PHP, I'll be using random_int() function or the random_bytes() function, which are specifically designed for cryptographic purposes and provide a higher level of randomness.

To generate a random number within a specified range:
$randomNumber = random_int($min, $max);
 
To generate random bytes for cryptographic purposes.
$randomBytes = random_bytes($length);

**Comment by dregad** [comment]
Thanks for your report. Please note

- adodb-time.inc.php was deprecated in 5.22.6 and removed in 5.23.0, see #970 
- not all usage of rand() actually requires a CSPRNG... your list contains mostly false positives

At first glance, only usages around ADOdb sessions (crypt.inc.php, adodb-session*.php are potentially problematic. 

This code is very old and shows its age... An in-depth refactoring would be needed, replacing *rand()* by *random_int()* would only scratch the surface and be kind of pointless, considering we have code performing md5 / sha1 "encryption", or relying on mcrypt (removed in PHP 7.2).

**Comment by DanielRuf** [comment]
@mariobernheim not really since you can see there, that a native PHP function is used, if available:

https://github.com/ADOdb/ADOdb/blob/e7256adf5f5f67e412807edabc34527e8387cb53/session/adodb-session2.php#L63
https://github.com/ADOdb/ADOdb/blob/e7256adf5f5f67e412807edabc34527e8387cb53/session/adodb-session.php#L77

This is just really old code with the `md5` logic, the `session_regenerate_id` function exists since PHP 4.3.2:
https://www.php.net/manual/en/function.session-regenerate-id.php

> (PHP 4 >= 4.3.2, PHP 5, PHP 7, PHP 8)

Basically this was not a real pentest, just a static code scan without verifying the findings with actual Proof of Concept examples.

So yes, there is an old (19+ years, see the blame view at https://github.com/ADOdb/ADOdb/blame/e7256adf5f5f67e412807edabc34527e8387cb53/session/adodb-session.php#L79) insecure logic.
But it is only the fallback, if the mentioned function is not available.

Also since it is not encryption but one-way hashing, it is a bit harder to get a collision.
